flight-server/src/main/java/org/m4d/adp/flight_server/RelayProducer.java

```diff
@@ -26,33 +36,40 @@
     private boolean transform = false;
     private final FlightClient client;
     private final Location location;
-    private final BigIntVector constVec;
-    private int RecordsPerBatch = 1024*1024;
-
-    private BigIntVector getConstIntVector() {
-        BigIntVector c = new BigIntVector("a", allocator);
-        for (int j = 0; j < this.RecordsPerBatch; j++) {
-            c.setSafe(j, 13);
-        }
-        c.setValueCount(this.RecordsPerBatch);
-        return c;
-    }
 
     public RelayProducer(Location location, Location remote_location,
                           BufferAllocator allocator, boolean transform) {
         this.transform = transform;
         this.allocator = allocator;
-        this.constVec = getConstIntVector();
         this.location = location;
         this.client = FlightClient.builder()
                 .allocator(allocator)
                 .location(remote_location)
                 .build();
     }
 
-    private VectorSchemaRoot transformVectorSchemaRoot(VectorSchemaRoot v) {
-        v = v.removeVector(0);
-        return v.addVector(0, this.constVec);
+    private byte[] WASMTransformVectorSchemaRoot(long instance_ptr, byte[] recordBatchByteArray) {
```

Comment:
This name is quite confusing, given that this function no longer works on a VectorSchemaRoot instance.
It might be a good idea to keep a WASMTransformVectorSchemaRoot() function which will get a VSR and return a VSR, and will call this function (send a byte[] and get a byte[]).